lib/util/geocoder.js

```diff
@@ -1,9 +1,126 @@
-import * as pelias from './geocoders/pelias'
+import * as arcgis from '@conveyal/geocoder-arcgis-geojson'
+import lonlat from '@conveyal/lonlat'
+import * as pelias from 'isomorphic-mapzen-search'
+import memoize from 'lodash.memoize'
 
-function getGeocoder (gcConfig) {
-  return pelias
+/**
+ * Helper for creating a setting up a custom response rewrite function for a
+ * certain api response.
+ */
+function makeResponseRewriter (rewriters, key) {
+  return response => {
+    if (rewriters && typeof rewriters[key] === 'function') {
+      return rewriters[key](response)
+    }
+    return response
+  }
 }
 
-export function reverse (point, gcConfig) {
-  return getGeocoder(gcConfig).reverse(point, gcConfig)
+/**
+ * Create customized geocoder funcitons given a certain geocoding API, the
+ * config for the geocoder and response rewrite functions specific to this
+ * application. Any geocoder api that is added is expected to have an API that
+ * behaves very closely to https://github.com/conveyal/isomorphic-mapzen-search
+ */
+function makeGeocoder (geocoderApi, geocoderConfig, responseRewriters) {
+  const {apiKey, baseUrl, boundary, focusPoint} = geocoderConfig
+  return {
+    autocomplete: query => {
```

Comment:
Could you give each of these functions (autocomplete, getLocationFromGeocodedFeature) some basic comments on their behavior?